### DISCUSSION TITLE
Using Substitution Filters with the "Image" Comparison Method

### DISCUSSION THREAD

**Discussion by Yanwa** [WinMerge/winmerge#1893] [Q&A] [answered]
Hello,

So I'm trying to compare two PDFs to each other. Some of the things that I'm checking for are like font and verbiage so I was thinking that with the Image comparison method, I was able to get a diff report on showing these differences. However, I wasn't able to get the filters to work. Is there a way to get this to work or is there a better suited comparison method that I should be using? I was using the below regexs for working with the date and timestamp which I manage to get to work with the text comparison method but I couldn't get it to work with the image comparison method.

\d*/\d*/\d* and \d*:\d*

**Comment by sdottaka** [comment, marked as answer]
Unfortunately Substitution filters don't work for images.
To use Substitution filters, you need to extract and compare PDF text parts with Apache Tika plugin or xdocdiff WinMerge Plugin.

**Reply by Yanwa** [reply]
Thank you for the reply. I'll take a look at Apache Tika and xdocdiff.
